src/main/java/com/atlassian/db/replica/spi/state/StateListener.java

```diff
@@ -0,0 +1,7 @@
+package com.atlassian.db.replica.spi.state;
+
+import com.atlassian.db.replica.api.state.States;
+
+public interface StateListener {
```

Comment:
So far it seems it only has a no-op impl. What's the target use case?

Reply:
1. It's easier to test how states changes.
2. We can track transitions more gradually (for example, with the stack-trace mechanism) form the app.

I'm not 100% sure if we need this extra visibility available in API.